src/modules/esl-utils/dom/visible.ts

```diff
@@ -0,0 +1,48 @@
+import {getNodeName, getParentNode} from './api';
+
+export interface VisibilityOptions {
+  /** Element will be considered invisible if opacity set to '0' */
+  opacity?: boolean;
+  /** Element will be considered invisible if visibility set to 'hidden' */
+  visibility?: boolean;
+  /** Element will be considered invisible if it's not in viewport */
+  viewport?: boolean;
+  /** Element will be considered invisible if it's height or width is '0' */
+  size?: boolean;
+}
+
+/**
+ * Get the parent of the specified element with matching styles
+ * @param el - element for which to get the parent
+ * @param matchStyles - object of styles for comparing
+ */
+export function findParentByStyles(el: HTMLElement, matchStyles: Partial<CSSStyleDeclaration>): HTMLElement | undefined {
+  const styleNames = Object.keys(matchStyles);
+  if (!styleNames.length) return undefined;
+
+  const elStyle = window.getComputedStyle(el);
+  if (styleNames.some((styleName: any) => elStyle[styleName] === matchStyles[styleName])) return el;
```

Comment:
Not obvious point: we don't  declare that the operator-combinator is  OR
By default, if nothing special is mentioned, we assume that the combination will use AND operator 
e.g. find(el, {a: 1, b: 2}) means find element with property a:1 AND b: 2
We need to think how to deal with that conceptual problem (make it configurable, change check strategy or hide method from outside world)